test/e2e-tests/tests/pipelines_test.go

```diff
@@ -0,0 +1,112 @@
+package tests
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/opendatahub-io/ai-edge/test/e2e-tests/support"
+	"golang.org/x/net/context"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+)
+
+const (
+	AIEdgeE2EPipelineDirectoryRelativePath = "../../../pipelines/tekton/aiedge-e2e"
+	PipelineRunFileRelativePath            = AIEdgeE2EPipelineDirectoryRelativePath + "/aiedge-e2e.pipelinerun.yaml"
+)
+
+func CreateContext() context.Context {
+	return context.Background()
+}
+
+func WaitForAllPipelineRunsToComplete(ctx context.Context, clients *support.Clients) error {
```

Comment:
Actually here we can have a generic function waitFor() which would call this function and we don't need any sleeps here. We used something similar in the past with @sutaakar.
See here: https://github.com/kiegroup/kogito-operator/blob/main/test/pkg/framework/util.go#L83-L113
And more generic code here: https://gist.github.com/ngauthier/d6e6f80ce977bedca601

Reply:
Updated with the generic solution and tested 👍